x-pack/plugins/ml/server/models/results_service/build_anomaly_table_items.js

```diff
@@ -163,7 +168,7 @@ function aggregateAnomalies(anomalyRecords, interval, dateFormatTz) {
     }
     if (entitiesForDetector[entity] === undefined) {
       entitiesForDetector[entity] = record;
```

Comment:
Is `entity` guaranteed to be a safe value? If not, then we are potentially at risk here, too. I think there are three options available to you:
1) Use [`Object.defineProperty`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty) in place of `entitiesForDetector[entity]`.
2) Use a `Map` instead of an `Object` for `entitiesForDetector`.
3) Ensure that `entity` is a safe value.

Reply:
I believe this should be safe because we're checking for `entitiesForDetector[entity] === undefined` first, and the else if is checking for `Object.hasOwn(entitiesForDetector, entity)`